src/v/archival/archival_policy.cc

```diff
@@ -319,8 +328,13 @@ static ss::future<upload_candidate_with_locks> create_upload_candidate(
     auto result = ss::make_lw_shared<upload_candidate>(
       {.term = term, .sources = {segment}});
 
-    co_await get_file_range(
+    auto file_range_result = co_await get_file_range(
       begin_inclusive, end_inclusive, segment, result, io_priority);
+    if (file_range_result != storage::convert_to_file_pos_outcome::success) {
+        vlog(
+          archival_log.warn, "Failed to get file range: {}", file_range_result);
+        co_return upload_candidate_with_locks{upload_candidate{}, {}};
```

Comment:
Returning the blank object assumes caller knows how to handle it, maybe this function should return a std::optional so that we can return nullopt on the error case?

Reply:
yeah, we should use something like a variant here or boost outcome to let the caller know what failed, but this is just to fit into the current code structure. In the archiver service (which is the consumer of all of these calls) we check if the upload candidate is blank.

I think we should make changes to return type in a different PR to reduce the change surface area but happy to take a stab at it now if you think otherwise.